### PR TITLE
[FUSEQE-6253] Added api-key tests for api client

### PR DESCRIPTION
**Pull request by mmuzikar**
//test: `@api-client-auth`

### CODE REVIEW

File: ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPageObject.java [hunk 1]
```diff
@@ -62,7 +62,7 @@ public SelenideElement getButton(String buttonTitle, SelenideElement differentRo
             .filter(Condition.exactText(buttonTitle)).shouldHave(sizeGreaterThanOrEqual(1));
 
         if (foundButtons.size() > 1) {
-            fail("Ambiguous button title. Found more that 1 button with title " + buttonTitle);
+            log.warn("Ambiguous button title. Found more that 1 button with title {}", buttonTitle);
```
Comment (mmuzikar) [inline ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisPageObject.java]: Nasty, but Apicurio has two save buttons when editing API Client, but both do the same.

File: ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisRootPage.java [hunk 2]
```diff
@@ -15,7 +15,7 @@
     }
 
     private static final class Element {
-        public static final By ROOT = By.id("root");
+        public static final By ROOT = By.tagName("body");
```
Comment (mmuzikar) [inline ui-tests/src/test/java/io/syndesis/qe/pages/SyndesisRootPage.java]: The same issue, Apicurio doesn't use root id, but we might want to use some of our steps for apicurio as well.

**Comment by syndesisqe** [comment]
Checkstyle verification failed with:
```
ui-tests/src/test/java/io/syndesis/qe/steps/
  CommonSteps.java:1070:79: '{' is not preceded with whitespace. [WhitespaceAround]
ui-tests/src/test/java/io/syndesis/qe/steps/
  CommonSteps.java:1075:30: '{' is not preceded with whitespace. [WhitespaceAround]
ui-tests/src/test/java/io/syndesis/qe/steps/customizations/connectors/
  ApicurioSteps.java:377: Line has trailing spaces. [RegexpSingleline]
ui-tests/src/test/java/io/syndesis/qe/steps/customizations/connectors/
  ApicurioSteps.java:379:53: '{' is not preceded with whitespace. [WhitespaceAround]
ui-tests/src/test/java/io/syndesis/qe/steps/customizations/connectors/
  ApicurioSteps.java:381:33: '{' is not preceded with whitespace. [WhitespaceAround]
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by avano** [comment]
retest this please

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Testing authentication options of API client | API key used in API client provided in headers 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Testing authentication options of API client | API key used in API client provided in headers  - known issue
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Testing authentication options of API client | API key used in API client provided in headers  - known issue
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by avano** [comment]
@mmuzikar what is the status of this?

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Integration - Database | Smoke - Periodic invocation to Insert 
Testing authentication options of API client | API key used in API client provided in query 
Testing authentication options of API client | API key used in API client provided in headers  - known issue
Integration - DB to DB | Read and update operations 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
Some changes in Apicurio testsuite are needed, I am currently working on figuring those out.

**Comment by mmuzikar** [comment]
retest this please

**Comment by syndesisqe** [comment]
Following scenarios failed:
```
Testing authentication options of API client | API key used in API client provided in headers 
```

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mmuzikar** [comment]
retest this please

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Pull request merged**